R/get_ssm_by_sample.R

```diff
@@ -0,0 +1,117 @@
+#' @param maf_cols if basic_columns is set to FALSE, the user can specify what columns to be returned within the MAF. This parameter can either be a vector of indexes (integer) or a vector of characters.
+#' @param verbose Enable for debugging/noisier output.
+#'
+#' @return data frame in MAF format.
+#'
+#' @import dplyr tidyr
+#' 
+#' @export
+#'
+#' @examples
+#' #Basic usage:
+#' this_sample_df = get_ssm_by_sample(this_sample_id = "DOHH-2",
+#'                                    this_seq_type = "genome",
+#'                                    projection = "grch37")
+#'
+#' #Get meta
+#' dohh2_meta = GAMBLR.data::sample_data$meta %>% dplyr::filter(sample_id == "DOHH-2")
+#'
+#' #Get SSM with some filters applied:
+#' dohh2_ssm = get_ssm_by_sample(this_sample_id = "DOHH-2", 
+#'                               these_samples_metadata = dohh2_meta, 
+#'                               these_genes = c("MYC", "BCL2"), 
+#'                               verbose = FALSE,
+#'                               this_seq_type = "genome",
+#'                               projection = "grch37")
+#'
+get_ssm_by_sample = function(this_sample_id,
+                             this_seq_type,
+                             these_samples_metadata,
+                             tool_name = NULL,
+                             projection = "grch37",
+                             these_genes,
+                             augmented = NULL,
+                             flavour = NULL,
+                             min_read_support = 3,
+                             basic_columns = TRUE,
+                             maf_cols = NULL,
+                             verbose = FALSE){
+  
+  #warn/notify the user what version of this function they are using
+  message("Using the bundled SSM calls (.maf) calls in GAMBLR.data...")
+  
+  #get invalid parameters for this function
+  invalid_params = c("tool_name", "augmented", "flavour")
+  
+  #check if any such parameters are provided
+  for(param in invalid_params){
+    if(!is.null(get(param))){
+      print(paste("Unsupported parameter supplied. This is only available in GAMBLR.results:", param))
+      stop()
+    }
+  }
+  
+  #figure out which unix_group this sample belongs to
+  if(missing(these_samples_metadata)){
+    these_samples_metadata = GAMBLR.data::get_gambl_metadata(seq_type_filter = this_seq_type) %>%
+      dplyr::filter(sample_id == this_sample_id)
+    
+  }else{
+    these_samples_metadata = these_samples_metadata %>%
+      dplyr::filter(sample_id == this_sample_id)
+  }
+  
+  #get valid projections
+  valid_projections = grep("meta", names(GAMBLR.data::sample_data), value = TRUE, invert = TRUE)
+  
+  #return SSMs based on the selected projection
+  if(projection %in% valid_projections){
+    sample_ssm = GAMBLR.data::sample_data[[projection]]$maf %>%
+      dplyr::filter(Tumor_Sample_Barcode == this_sample_id)
```

Comment:
This error comes up a lot. Your tests/examples don't catch this error. This will fail if the length of this_sample_id is >1.

Reply:
Fixed